util/collate/ucadata/generator/data.go.tpl

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by "util/collate/ucadata/generator"; DO NOT EDIT.
+
+package ucadata
+
+// {{.Name}} is the collation element table for the corresponding unicode version.
+// it's generated from https://www.unicode.org/Public/UCA/{version}/allkeys-{version}.txt
```

Comment:
{{.version}}

Reply:
I found the url pattern of 4.0.0 and 9.0.0 is different. One of them include version in file name and the other doesn't, so I use a `{{.Url}}` directly instead. Nice catch.